src/components/search/UseSearch.jsx

```diff
@@ -31,16 +33,26 @@ const useSearch = () => {
     } catch (error) {
       console.error('Failed to fetch classes:', error.message);
     } finally {
-      setLoading(false);
+      setTimeout(() => {
+        setLoading(false);
+      }, 200);
     }
   };
+  //this is the way it works for both MainPage(DesktopNav) and SearchPage page
+
+  useEffect(() => {
+    if (search) {
+      fetchClasses(search, currentPage);
+    }
+  }, [search, currentPage, category]);
 
   useEffect(() => {
     fetchClasses(search, currentPage);
   }, [search, currentPage, category]);
 
   const handleSearch = (searchTerm) => {
```

Comment:
`handleSearch` is being thrown around a lot. Be careful. I suggest refactoring to only use this function were necessary. Because it is part of a hook you have given this function a lot of flexibility which can lead to some confusion as to were and how it should be implemented. Additionally, you search components have a lot a useEffects that are probably triggering more often than intended. Also, reconsider how you are using those functions.